### PR TITLE
Fix data race in `therealssh`

### LINKED ISSUE #411: Race condition: therealssh
```
==================
WARNING: DATA RACE
Read at 0x00c0001402f0 by goroutine 18:
  github.com/skycoin/skywire/internal/therealssh.(*SSHChannel).Close()
      /home/travis/gopath/src/github.com/skycoin/skywire/internal/therealssh/channel.go:261 +0x13d
  github.com/skycoin/skywire/internal/therealssh.TestChannelServeSocket()
      /home/travis/gopath/src/github.com/skycoin/skywire/internal/therealssh/channel_test.go:125 +0xa45
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:865 +0x163
Previous write at 0x00c0001402f0 by goroutine 19:
  github.com/skycoin/skywire/internal/therealssh.(*SSHChannel).ServeSocket.func1()
      /home/travis/gopath/src/github.com/skycoin/skywire/internal/therealssh/channel.go:162 +0x8f
  github.com/skycoin/skywire/internal/therealssh.(*SSHChannel).ServeSocket()
      /home/travis/gopath/src/github.com/skycoin/skywire/internal/therealssh/channel.go:177 +0x7c9
  github.com/skycoin/skywire/internal/therealssh.TestChannelServeSocket.func1()
      /home/travis/gopath/src/github.com/skycoin/skywire/internal/therealssh/channel_test.go:101 +0x38
Goroutine 18 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:916 +0x65a
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:1157 +0xa8
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:865 +0x163
  testing.runTests()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:1155 +0x523
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:1072 +0x2eb
  main.main()
      _testmain.go:122 +0x334
Goroutine 19 (finished) created at:
  github.com/skycoin/skywire/internal/therealssh.TestChannelServeSocket()
      /home/travis/gopath/src/github.com/skycoin/skywire/internal/therealssh/channel_test.go:101 +0x3e3
  testing.tRunner()
      /home/travis/.gimme/versions/go1.12.6.linux.amd64/src/testing/testing.go:865 +0x163
==================
--- FAIL: TestChannelServeSocket (0.30s)
    testing.go:809: race detected during execution of test
FAIL
```

### PR DESCRIPTION
**Pull request by Darkren**
Fixes #411 .

The problem was that the `listener` was not protected from concurrent `read/write` operations. Added mutex to avoid it and wrapped closing into `closeListener`

### CODE REVIEW

**Review by nkryuchkov — APPROVED** [review]
Looks good to me

File: internal/therealssh/channel.go [hunk 1]
```diff
@@ -273,6 +274,19 @@ func (sshCh *SSHChannel) Close() error {
 	return nil
 }
 
+func (sshCh *SSHChannel) closeListener() error {
+	sshCh.listenerMx.Lock()
+	defer sshCh.listenerMx.Unlock()
+
+	if sshCh.listener != nil {
```
Comment (ayuryshev) [inline internal/therealssh/channel.go]: This check is already embedded in `UnixListener`.

Look at https://github.com/golang/go/blob/9aa2c06cb5cb4603500e5fbc35385fe890c6779b/src/net/unixsock.go#L227
And `UnixListener.Close()` calls this check.

So it's safe to call sshCh.listener.Close() without checkя

**Review by ayuryshev — APPROVED** [review]

**Comment by evanlinjin** [comment]
Also, please resolve conflicts.

**Comment by jdknives** [comment]
@Darkren please resolve and then it seems ready to merge.

**Comment by Darkren** [comment]
@jdknives resolved

**Comment by jdknives** [comment]
@Darkren I re-requested review from @ayuryshev @nkryuchkov but any one of their  reviews should suffice. This has been tested in the interactive environment?

**Comment by Darkren** [comment]
@jdknives This was fairly a time ago, I'd better retest it shortly

**Comment by Darkren** [comment]
@jdknives Tested, seems to work

**Review by nkryuchkov — APPROVED** [review]

**Comment by ayuryshev** [comment]
Looks good

**Review by ayuryshev — APPROVED** [review]

**Pull request merged**